src/pages/index.js

```diff
@@ -0,0 +1,305 @@
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardDeleteButton.addEventListener("click", (evt) =>
+    handleDeleteCardButton(cardElement, data._id)
+  );
+
+  return cardElement;
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function handleClickOffClose(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscClose);
+  modal.addEventListener("click", handleClickOffClose);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+  modal.removeEventListener("click", handleClickOffClose);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false, "Delete", "Deleting...");
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((user) => {
+      profileName.textContent = user.name;
+      profileDescription.textContent = user.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement({
+        ...inputValues,
+        _id: data._id,
+        isLiked: false,
+      });
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .editUserAvatar({ avatar: avatarModalInput.value })
+    .then((user) => {
+      avatarElement.src = user.avatar;
+      avatarElement.alt = user.name;
+      evt.target.reset();
+      disableButton(avatarSubmitButton, settings);
```

Comment:
`avatarSubmitButton`


* When I change the `avatar`, there is an error https://snipboard.io/Z0yrnP.jpg 

`avatarSubmitButton` doesn't exist. You should use `submitButton` from line 241


```js

      disableButton(submitButton, settings);

```